app/assets/javascripts/application.js

```diff
@@ -14,3 +14,4 @@
 //= require jquery_ujs
 //= require turbolinks
 //= require_tree .
+//= require upload
```

Comment:
I think that `require_tree .` will grab `upload.js`.  Is there a reason you're calling out `require upload` specifically?